.circleci/config.yml

```diff
@@ -134,14 +136,20 @@ redis-container: &redis-container
 
 dnsmasq-container: &dnsmasq-container
   image: quay.io/3scale/dnsmasq
+  entrypoint:
+    - /bin/bash
+    - -c
   command:
-    - --user=root
-    - --keep-in-foreground
-    - --log-facility=-
-    - --log-queries
-    - --no-poll
-    - --server=/circleci-internal-outer-build-agent/circleci-internal-outer-build-agent.ec2.internal/circleci.com/circleci-binary-releases.s3.amazonaws.com/127.0.0.11
-    - --address=/#/127.0.0.1
+    - |
+      DEFAULT_DNS=$(sed -nEe 's/nameserver\s+(.*)/\1/p' /etc/resolv.conf | head -1)
+      /usr/sbin/dnsmasq \
+      --user=root \
+      --keep-in-foreground \
+      --log-facility=- \
+      --log-queries \
+      --no-poll \
+      --server=/circleci.com/dns.podman/circleci-tasks-prod.s3.us-east-1.amazonaws.com/circleci-binary-releases.s3.amazonaws.com/$DEFAULT_DNS \
```

Comment:
@akostadinov what are these URLs?... why do we need them?

Reply:
CircleCI cli downloads some plugins or whatever from some of them. Also the reporting of tests and artifacts happens over the network. So if these are not properly resolved, some or all of these fail. For example in the past we had issues reporting failed tests because of this.

They can be discovered by inspecting the dnsmasq log where all queries and their resolutions are reported.

Reply:
btw I had a PM conversation with circleci support and I asked them to check if we have something non-future-proof. Hope they answer next week.